### PR TITLE
[TM ONLY] Live debugging hooks for misdelivered chat messages.

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
TM ONLY
Adds temporary debugging hooks for misdelivered chat messages.

## Why It's Good For The Game
We've had a sporadic bug for quite a while, where occasionally a chat message you should hear goes missing, or you hear one you definitely shouldn't, like non-admins being told their admin verbs were hidden.

So far, it's a mystery whether this is a bug in our code or BYOND, though I suspect at least some of it is on our end. This debug info will help me track down what's happening, assuming it's not just a pure BYOND bug.

## Testing
Started server. Heard chat normally.
Put in a setting I could varedit to make all messages "misdeliver", successfully heard misdelivery notices.

## Changelog
NPFC that should not be merged.

### CODE REVIEW

**Comment by Danchi299** [comment]
make changelog say 
```
:cl:
add: if you see this something went horribly wrong
/:cl:
```
😄

File: code/datums/chat_payload.dm [hunk 1]
```diff
@@ -6,6 +6,9 @@
 	var/list/content
 	/// Resend count
 	var/resends = 0
+	var/target
+	var/list/targets
```
Comment (SteelSlayer) [inline code/datums/chat_payload.dm:10]: I think these references need to be cleared on destroy

File: code/datums/chat_payload.dm [hunk 2]
```diff
@@ -6,6 +6,14 @@
...
+	var/delivery_attempt = 0
+
+/datum/chat_payload/Destroy(force = FALSE, ...)
+	. = ..()
+	target = null
+	targets.Cut()
```
Comment (GDNgit) [inline code/datums/chat_payload.dm:16]:
```suggestion
	target = null
	targets.Cut()
	return ..()
```

**Comment by FunnyMan3595** [comment]
No luck, sadly.

**Pull request closed without merging**